File: VB-Examples/07_Conversion/ToHtmlWithHiddenWorksheets.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace ToHtmlWithHiddenWorksheets
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of the Workbook class.
            Dim book As New Workbook()

            ' Load the Excel file from the specified path.
            book.LoadFromFile("..\..\..\..\..\..\Data\ToHtmlWithHiddenWorksheets.xlsx")

            ' Specify the output HTML file name.
            Dim result As String = "result.html"

            ' Save the workbook to an HTML file.
            ' Parameter: 
            ' - False: Save to HTML with hidden worksheets included.
            ' - True: Save to HTML without hidden worksheets.
            book.SaveToHtml(result, False)
            ' Release the resources used by the workbook
            book.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/07_Conversion/ToHtmlWithHiddenWorksheets.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace ToHtmlWithHiddenWorksheets
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook book = new Workbook();

            // Load the document
            book.LoadFromFile(@"..\..\..\..\..\..\Data\ToHtmlWithHiddenWorksheets.xlsx");
			
			// Save Excel to Html
			// false --- To Html with the hidden Worksheet
            // true--- To Html without the hidden Worksheet
            string result = "result.html";
            book.SaveToHtml(result, false);


			// Dispose of the workbook object to release resources
			book.Dispose();
			
            // Launch the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
